### PR TITLE
cli: do not require --containers for set triggers on dc with one container

### PR DESCRIPTION
**Pull request by mfojtik**
Fixes: https://github.com/openshift/origin/issues/15185

[test]

@kargakis PTAL

### CODE REVIEW

File: pkg/cmd/cli/cmd/set/triggers.go [hunk 1]
```diff
@@ -656,6 +656,11 @@ func (t *TriggerDefinition) Apply(obj runtime.Object) error {
 			allNames.Insert(container.Name)
 		}
 		for _, trigger := range t.ImageChange {
+			// If we have only one container, assume the triggers should be set for that
+			// container and do not require --containers to be specified.
+			if len(allNames) == 1 {
+				trigger.Names = append(trigger.Names, allNames.List()[0])
+			}
 			if len(trigger.Names) == 0 {
 				return fmt.Errorf("you must specify --containers when setting --from-image")
```
Comment (kargakis) [inline pkg/cmd/cli/cmd/set/triggers.go]: Should we also update this error message?

Comment (mfojtik) [inline pkg/cmd/cli/cmd/set/triggers.go]: why? if allNames > 1 this error is valid?

Comment (kargakis) [inline pkg/cmd/cli/cmd/set/triggers.go]: The error message must contain that `-c` is required when multiple containers exist in a template.
"you must specify --containers when using --from-image with multiple containers"

Comment (mfojtik) [inline pkg/cmd/cli/cmd/set/triggers.go]: gotcha, sure

**Comment by mfojtik** [comment]
@kargakis fixed, i even added hint what containers are present (because naturally the next thing I will do after getting this message would be to lookup what are the container names ;-)

**Comment by kargakis** [comment]
Awesome, lgtm

We are during stage cut if I am not mistaken?

**Comment by mfojtik** [comment]
[merge]

no, merge freeee!

**Comment by smarterclayton** [comment]
Test is broken

File: pkg/cmd/cli/cmd/set/triggers.go [hunk 2]
```diff
@@ -656,8 +656,13 @@ func (t *TriggerDefinition) Apply(obj runtime.Object) error {
 			allNames.Insert(container.Name)
 		}
 		for _, trigger := range t.ImageChange {
+			// If we have only one container, assume the triggers should be set for that
+			// container and do not require --containers to be specified.
+			if len(allNames) == 1 {
+				trigger.Names = []string{allNames.List()[0]}
```
Comment (Miciah) [inline pkg/cmd/cli/cmd/set/triggers.go]: If `--containers` is specified and identifies a container that does not exist, will this silently override the user's specification?

Comment (juanvallejo) [inline pkg/cmd/cli/cmd/set/triggers.go]: We could output a warning to stderr when this is the case, to at least inform the user that the specified container was not found.

Alternatively, we could just ignore this conditional if `--containers` was specified by the user, and fail as usual.

cc @fabianofranz

**Comment by juanvallejo** [comment]
lgtm aside from @miciah 's comment

**Comment by kargakis** [comment]
@mfojtik needs rebase

**Pull request closed without merging**